source/common/router/router.cc

```diff
@@ -156,6 +156,13 @@ FilterUtility::finalTimeout(const RouteEntry& route, Http::RequestHeaderMap& req
     } else {
       timeout.global_timeout_ = route.timeout();
     }
+  } else {
```

Comment:
When we switched timeouts, we moved the instantiation of the timeout to
ConnectionManagerImpl::ActiveStream::refreshDurationTimeout()

Mind moving this logic there?